gr-digital/lib/constellation.cc

```diff
@@ -297,7 +347,16 @@ std::vector<float> constellation::calc_soft_dec(gr_complex sample, float npwr)
     // probability of ones (tmp[2*i+1]) over the probability of a zero
     // (tmp[2*i+0]).
     for (int i = 0; i < k; i++) {
-        s[k - 1 - i] = (logf(tmp[2 * i + 1]) - logf(tmp[2 * i + 0]));
+        float one = tmp[2 * i + 1];
+        float zero = tmp[2 * i + 0];
+        // clamp input to log to prevent NaN and Inf.
+        if (one < 3.4e-38) {
+            one = 3.4e-38;
+        }
+        if (zero < 3.4e-38) {
+            zero = 3.4e-38;
+        }
```

Comment:
I'm curious about were the `3.4e-38` came from. My numpy is fine down to `1e-45`:
```python
>>> np.log(np.float32(1e-45))
-103.27893
```

Also, I wonder whether we hit this in practice. For this to happen, the sum of probabilities (for `one` or for `zero`) would need to be very close to zero. For this to happen, since those probabilities are the PDF of a gaussian, that would happen if the Euclidean distance from `sample` to all the constellation points corresponding to a bit value is larger than about 13 `noise_sigma` (because `exp(-90) = 8.19e-40` and `sqrt(2 * 90) = 13.4`). I guess this isn't impossible if one sets `noise_sigma` to a small value (i.e., if one expects a very high SNR), so it's good to have this check.

Reply:
Yes, this is exactly why I put this check. Even modest snrs above 13db or so will begin producing spurious values. The odd clamp value was the float32 precision limit off the top of my head. Can be changed, just works.

Reply:
Your change works :)